src/GafferScene/SceneWriter.cpp

```diff
@@ -189,6 +189,27 @@ void SceneWriter::writeLocation( const GafferScene::ScenePlug *scene, const Scen
 		output->writeTransform( new IECore::M44dData( transform ), time );
 	}
 
+	// query the set membership for this location and write as tags.
+	const std::vector<InternedString>& setNames = scene->setNamesPlug()->getValue()->readable();
+
+	SceneInterface::NameList locationSets;
+	locationSets.reserve( setNames.size() );
+
+	for( size_t i = 0, numSets = setNames.size(); i < numSets; ++i )
+	{
+		context->set( ScenePlug::setNameContextName, setNames[i] );
```

Comment:
Great! I was considering caching the sets but wasn't sure if this was wrong headed in terms of how contexts work and how gaffer caches. Thanks for the explanation.